pkg/hosts/hosts.go

```diff
@@ -88,57 +89,85 @@ func (h *Hosts) Add(ipRaw string, hosts []string) error {
 		return err
 	}
 
-	line, err := h.findIP(start, end, ip)
+	lines, err := h.findIP(start, end, ip)
 	if err != nil {
 		return err
 	}
 
 	// no host record, need to create new host line
-	if line == nil {
-		hfl := libhosty.HostsFileLine{
-			Type:      libhosty.LineTypeAddress,
-			Address:   ip,
-			Hostnames: hostEntries,
+	if lines == nil {
+		if len(hostEntries) > maxHostsInLine {
+			multiLineHosts := ChunkSlice(hostEntries, maxHostsInLine)
+			for _, hosts := range multiLineHosts {
+				h.createAndAddHostsLine(ip, hosts, start)
+			}
+		} else {
+			h.createAndAddHostsLine(ip, hostEntries, start)
 		}
 
-		// inserts to hosts
-		newHosts := make([]libhosty.HostsFileLine, 0)
-		newHosts = append(newHosts, h.File.HostsFileLines[:start+1]...)
-		newHosts = append(newHosts, hfl)
-		newLineNum := len(newHosts) - 1
-		h.File.Lock()
-		newHosts = append(newHosts, h.File.HostsFileLines[start+1:]...)
-		h.File.HostsFileLines = newHosts
-
-		// generate raw version of the line
-		hfl.Raw = h.File.RenderHostsFileLine(newLineNum)
-		h.File.Unlock()
-
 	} else {
 		var hostToAdd []string
 		for _, hostName := range hostEntries {
 			// check that new host not present in this line
 			contains := false
-			for _, lineHost := range line.Hostnames {
-				if hostName == lineHost {
-					contains = true
-					break
-				}
-			}
+
+			lineNum, _ := h.File.GetHostsFileLineByHostname(hostName)
+
+			// check that line inside crc region
+			contains = lineNum > start && lineNum < end
+
 			// add only new hosts
 			if !contains {
 				hostToAdd = append(hostToAdd, hostName)
 			}
```

Comment:
This could go to a helper `hostToAdd := filterExistingEntries(h, hostEntries)`
The `Add` method is very long with some slightly tricky code, so I'm trying to find ways to have smaller functions which are easier to review for correctness, and help with `Add` readability.